### PR TITLE
Add Helm chart for MongoDB sharded cluster

### PR DESCRIPTION
**Pull request by sejongk**
**What this PR does / why we need it**:
This PR adds a Helm chart for MongoDB, supporting both standalone mode and sharded cluster mode.
Shard rules, including shard keys, shard methods, and unique constraints, are defined and managed in the `values.yaml`.

This chart depends on the Bitnami's `mongodb-sharded` chart (https://github.com/bitnami/charts/tree/main/bitnami/mongodb-sharded).
There are a few known issues with it:
- A problem with livenessProbe and readinessProbe
  - Issues
    - https://github.com/bitnami/charts/issues/21957
  - Solution: use custom livenessProbe and readinessProbe instead
- Lack of ARM64 support in the Bitnami `mongodb-sharded` **container**
  - Issues 
    - https://github.com/bitnami/charts/issues/7305#issuecomment-1443187343
    - https://github.com/bitnami/containers/issues/40947
  - Solution: use the official mongo 6.0 image instead and set up the cluster via a Job. 

### How to install and uninstall
1. Create `mongodb` namespace
```
kubectl create namespace mongodb
```
2. Install helm chart with the set flag and the namespace flag (The reason why the namespace flag is needed to be set is due to the misguided value in the Bitnami chart. In [here](https://github.com/bitnami/charts/blob/947458bc7c22d489a506306ba48782d38b906450/bitnami/mongodb-sharded/templates/_helpers.tpl#L36), it uses `.Release.Namespace` without offering an option to override it. 
```
helm install mongodb . --namespace mongodb --set=sharded.enabled=true
```
3. Uninstall helm chart with the namespace flag
```
helm install mongodb . --namespace mongodb
```
**Which issue(s) this PR fixes**:

Addresses #673 

**Special notes for your reviewer**:

We need to do following tasks in the future:

- [ ] Enable livenessProbe, readinessProbe
- [ ] Connect to Prometheus via `PodMonitor`s

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: build/charts/yorkie-mongodb/templates/sharded/configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          {{- range .shardKeys }}
+            {{ printf "%s: %s," .name .method }}
+          {{- end }}
+        }, 
+        {{ .unique }}
+      );
+      {{- end }}
+    EOF
+    
+
```
Comment (hackerwins) [inline build/charts/yorkie-mongodb/templates/sharded/configmap.yaml:69]:
```suggestion
    
```

File: build/charts/yorkie-mongodb/values.yaml [hunk 2]
```diff
@@ -0,0 +1,189 @@
+namespace: &namespace mongodb
+name: &name mongodb
+port: &port 27017
+clusterDomain: &clusterDomain cluster.local
+
+image:
```
Comment (hackerwins) [inline build/charts/yorkie-mongodb/values.yaml:6]: Could you add descriptions for each section(image, persistence, sharded, and mongodb-sharded)?

**Review by hackerwins — CHANGES_REQUESTED** [review]
Thanks for your contribution.
I left a few comments.

**Review by krapie — COMMENTED** [review]
Thank you for your contribution 😄
First off, I have left some small conventional suggestions along with comprehensive questions.

#### Question 1

It seems like `yorkie-cluster` Helm chart no longer has mongoDB subshart. This means that we now need to install both `yorkie-cluster` and `yorkie-mongodb` Helm chart to fully use yorkie cluster. Is there any reason for this separation?

#### Question 2

It seems like `yorkie-mongodb` Helm chart performs sharded mongoDB setup by using Job object. Is there any reason for this approach?

I can see that mongodb-sharded Helm chart provides `common.initScriptsCM` value to initialize mongoDB instances (Nevertheless [I also think using Job is the best approach for this kind of operation](https://github.com/docker-library/mongo/issues/339#issuecomment-1276780427), I just want to know the reasons for this approach).

> Lack of ARM64 support in the Bitnami mongodb-sharded container
> Solution: use the official mongo 6.0 image instead and set up the cluster via a Job.

Also, is there any relations between ARM64 support and setting up cluster via Job?

#### Suggestion 3

How about adding [`NOTES.txt`](https://github.com/yorkie-team/yorkie/blob/main/build/charts/yorkie-cluster/templates/NOTES.txt) to keep consistency with other yorkie Helm charts?

#### Suggestion 4

We might need to add [ArgoCD application manifest](https://github.com/yorkie-team/yorkie/blob/main/build/charts/yorkie-argocd/templates/applications.yaml) for `yorkie-mongodb` in `yorkie-argocd` Helm chart.

**Review by krapie — APPROVED** [review]
LGTM 👍🏼

**Review by hackerwins — APPROVED** [review]
LGTM.

**Comment by sejongk** [comment]
> #### Question 1
> It seems like `yorkie-cluster` Helm chart no longer has mongoDB subshart. This means that we now need to install both `yorkie-cluster` and `yorkie-mongodb` Helm chart to fully use yorkie cluster. Is there any reason for this separation?
> 
> #### Question 2
> It seems like `yorkie-mongodb` Helm chart performs sharded mongoDB setup by using Job object. Is there any reason for this approach?
> 
> I can see that mongodb-sharded Helm chart provides `common.initScriptsCM` value to initialize mongoDB instances (Nevertheless [I also think using Job is the best approach for this kind of operation](https://github.com/docker-library/mongo/issues/339#issuecomment-1276780427), I just want to know the reasons for this approach).
> 
> > Lack of ARM64 support in the Bitnami mongodb-sharded container
> > Solution: use the official mongo 6.0 image instead and set up the cluster via a Job.
> 
> Also, is there any relations between ARM64 support and setting up cluster via Job?

Thanks for your careful review.
Here are my answers for your questions.

#### Answer 1
After separating MongoDB from yorkie-cluster, there are now two options for users: self-hosted or managed databases (e.g., MongoDB Atlas, AWS DynamoDB). I believe we can focus on supporting self-hosted clusters by solely maintaining yorkie-mongodb.

#### Answer 2
I initially explored the approach using Bitnami options and tools. Bitnami charts are designed to be used with Bitnami containers together. However, issues arose, including the lack of ARM64 images in the Bitnami `mongodb-sharded` container. Instead of incurring the current and potential future engineering costs associated with external tools, I opted to implement and maintain the code ourselves. 
I believe the current requirements for this are relatively simple and manageable. If future requirements become more complex and there is a good alternative, that would be the time to consider such an option.

**Pull request merged**